attribution-data-file-share/src/main/java/gov/cms/ab2d/attributionDataShare/AttributionDataShareHandler.java

```diff
@@ -0,0 +1,87 @@
+package gov.cms.ab2d.attributionDataShare;
+
+import com.amazonaws.services.lambda.runtime.Context;
+import com.amazonaws.services.lambda.runtime.LambdaLogger;
+import com.amazonaws.services.lambda.runtime.RequestStreamHandler;
+import gov.cms.ab2d.lambdalibs.lib.PropertiesUtil;
+
+import java.io.*;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Properties;
+
+public class AttributionDataShareHandler implements RequestStreamHandler {
+
+    private static final String FILE_PATH = "/opt/";
+    private static final String FILE_PARTIAL_NAME = "ab2d-beneids_";
+    private static final String FILE_FORMAT = ".txt";
+
+    private static final String SELECT_ALL_FROM_COVERAGE = "SELECT current_mbi FROM public.coverage";
+
+    // Writes out a file to the FILE_PATH.
+    // I.E: "ab2d-beneids_2023-08-16T12:08:56.235-0700.txt"
+    public void handleRequest(InputStream inputStream, OutputStream outputStream, Context context) throws IOException {
+        LambdaLogger logger = context.getLogger();
+        logger.log("AttributionDataShare Lambda is started");
+
+        String currentDate = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSZ").format(new Date());
+        String fileFullPath = FILE_PATH + FILE_PARTIAL_NAME + currentDate + FILE_FORMAT;
+
+        try (FileWriter fileWriter = new FileWriter(fileFullPath)) {
+            Connection dbConnection = getConnection();
+
+            Statement statement = dbConnection.createStatement();
+            ResultSet resultSet = statement.executeQuery(SELECT_ALL_FROM_COVERAGE);
+
+            BufferedWriter bufferedWriter = new BufferedWriter(fileWriter);
+
+            if (resultSet.next() == false) {
+                logger.log("ResultSet is empty...skipping writing to file");
+            } else {
+                logger.log("Writing to file...");
+                do {
+                    String result = resultSet.getString(1);
+                    if (result != null || result != "") {
```

Comment:
I am not sure. I feel like resultSet.getString() could return "" at least. This condition was mostly here for safety but if it is truly not needed I can remove it. 

Does resultSet.getString() guarentee a non-null, non-empty string value?

Reply:
you checked that resultSet is not empty and than do while loop until there is smth in resultSet. It can't be null.

prod doesn't have rows with empty current_mbi

Reply:
I think I will leave the check for "" (empty string) just in case? Unless this can also be removed, but I have removed the check for not null because your explanation as to why it cannot be null makes sense.